clients/js/test/extensions/interestBearingMint/getAmountToUiAmount.test.ts

```diff
@@ -0,0 +1,296 @@
+    }
+    if (!(account.data instanceof Uint8Array)) {
+      throw new Error(
+        `Account data is not a Uint8Array for address: ${address}`
+      );
+    }
+    return {
+      send: async () => ({
+        context: { slot: 0n },
+        value: account
+          ? {
+              executable: account.executable,
+              lamports: account.lamports,
+              owner: account.owner,
+              rentEpoch: account.rentEpoch,
+              data: [
+                Buffer.from(account.data).toString('base64'),
+                'base64',
+              ] as [Base64EncodedBytes, 'base64'],
+            }
+          : null,
+      }),
+    };
+  };
+  return { getAccountInfo } as unknown as Rpc<GetAccountInfoApi>;
+}
+
+function populateMockAccount(data: Buffer) {
+  return {
+    executable: false,
+    lamports: 1000000n as Lamports,
+    owner: TOKEN_2022_PROGRAM_ADDRESS,
+    rentEpoch: 0n,
+    data,
+  };
+}
+
+function createMockMintAccountInfo(
+  decimals = 2,
+  hasInterestBearingConfig = false,
+  config: { preUpdateAverageRate?: number; currentRate?: number } = {}
+) {
+  const defaultAddress = address('11111111111111111111111111111111');
+  const mintEncoder = getMintEncoder();
+  const bufferData = Buffer.from(
+    mintEncoder.encode({
+      mintAuthority: defaultAddress,
+      supply: BigInt(1000000),
+      decimals: decimals,
+      isInitialized: true,
+      freezeAuthority: defaultAddress,
+      extensions: hasInterestBearingConfig
+        ? [
+            {
+              __kind: 'InterestBearingConfig',
+              rateAuthority: defaultAddress,
+              initializationTimestamp: BigInt(0),
+              preUpdateAverageRate: config.preUpdateAverageRate || 500,
+              lastUpdateTimestamp: BigInt(ONE_YEAR_IN_SECONDS),
+              currentRate: config.currentRate || 500,
+            },
+          ]
+        : [],
+    })
+  );
+  return populateMockAccount(bufferData);
+}
+
+const createMockClockAccountInfo = (unixTimestamp: number) => {
+  const clockEncoder = getSysvarClockEncoder();
+  const bufferData = Buffer.from(
+    clockEncoder.encode({
+      epoch: 0n,
+      epochStartTimestamp: BigInt(0) as UnixTimestamp,
+      leaderScheduleEpoch: 0n,
+      slot: 0n,
+      unixTimestamp: BigInt(unixTimestamp) as UnixTimestamp,
+    })
+  );
+  return populateMockAccount(bufferData);
+};
+
+const mint = address('So11111111111111111111111111111111111111112');
+const clock = SYSVAR_CLOCK_ADDRESS;
+
+test('should return the correct UiAmount when interest bearing config is not present', async (t) => {
+  const testCases = [
+    { decimals: 0, amount: BigInt(100), expected: '100' },
+    { decimals: 2, amount: BigInt(100), expected: '1' },
+    { decimals: 9, amount: BigInt(1000000000), expected: '1' },
+    { decimals: 10, amount: BigInt(1), expected: '1e-10' },
+    { decimals: 10, amount: BigInt(1000000000), expected: '0.1' },
+  ];
```

Comment:
We'd typically use macros in ava for this but don't worry about it because I'm thinking about using `vitest` for program clients in the future. So by leaving it like that, it'll make the refactoring much easier.